### PR TITLE
Send shutting down countdown during graceful shutdown period

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] https://github.com/ppy/osu/pull/20342

### CODE REVIEW

File: osu.Server.Spectator/GracefulShutdownManager.cs [hunk 1]
```diff
@@ -39,12 +44,36 @@ private void shutdownSafely()
...
+            });
+        }).Wait();
 
+        TimeSpan timeWaited = new TimeSpan();
         TimeSpan timeBetweenChecks = TimeSpan.FromSeconds(10);
+        bool finalNotificationSent = false;
 
         while (timeWaited < TIME_BEFORE_FORCEFUL_SHUTDOWN)
         {
+            TimeSpan timeRemaining = TIME_BEFORE_FORCEFUL_SHUTDOWN - timeWaited;
```
Comment (peppy) [inline osu.Server.Spectator/GracefulShutdownManager.cs:61]: Was this implementation chose due to ease of implementation? I would have thought that one countdown would ve sent to all rooms when shutdown started, and one would be sent to any users joining a room on a server which is shutting down (at the point of joining the room).

If that sounds too hard to implement we can go with what you have to start with.

Comment (smoogipoo) [reply osu.Server.Spectator/GracefulShutdownManager.cs:61]: What part? This is only done to send the other notification 5 minutes before shutdown. It is _also_ sent to all rooms at the start of the countdown and also when they join the room.

Should I remove the countdown in this loop? I'm unsure of it existing.

Comment (peppy) [reply osu.Server.Spectator/GracefulShutdownManager.cs:61]: I think just one notification type is fine (removing the boolean), as per discord discussion.

Ignore the rest of what I said, I missed that countdowns were sent as part of the room.

**Comment by peppy** [comment]
Will fix the warning level suggestions in a separate pull.

**Pull request merged**